crates/volta-core/src/layout/mod.rs

```diff
@@ -74,6 +75,7 @@ fn default_install_dir() -> Fallible<PathBuf> {
         .with_context(|_| ErrorDetails::NoInstallDir)
 }
 
+#[cfg(not(feature = "volta-updates"))]
```

Comment:
Can likely be a separate issue, and I think the "new" migration behavior fully encompasses the old behavior, since the old one only regenerated the config directory if the `~/.volta` folder didn't exist at all. That will still happen in this case, _and_ we can migrate from older to newer layouts.

The tricky part about any more involved "self-healing" behavior is that it requires a deeper inspection of the config directory, which we want to avoid on the hot path if possible.